WPMedia/fusion-news-theme-blocks
# TMEDIA-2 small promo code duplication

## Description

Make reusable promo headline and promo image components in shared styles to allow better reuse between small promo and small manual promo blocks. Also allow for other promo's to move over to consume them to reduce duplication of code between promo blocks.

Each of the reusable components can consume ANS content or custom content to allow them to work for both promo and manual promo blocks

In updating the code there are some slight visual differences with padding and gutters - will the acceptance criteria states no visual differences. The differences are fixing issues with the existing implementation of adding too much padding.


## Jira Ticket
- [TMEDIA-2](https://arcpublishing.atlassian.net/browse/TMEDIA-2)

## Acceptance Criteria
* Small Promo Block and Small Manual Promo Block should have no visible changes or visual regressions for the user
* Small Promo Block and Small Manual Promo Block should use a shared set of components to output promo content
* Small Promo Block custom fields remain the same via PageBuilder 
* Small Manual Promo Block custom fields remain the same via PageBuilder

## Test Steps

1. Checkout this branch `git checkout TMEDIA-2-small-promo-code-duplication`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/small-promo-block,@wpmedia/small-manual-promo-block,@wpmedia/shared-styles,@wpmedia/double-chain-block`
3. Add small promo and small manual promo blocks to a page and verify all still works as expected
4. Validate existing pages using small promo's still work as expected - http://localhost/pf/test-all-promos/?_website=the-gazette

## Effect Of Changes
### Before

![TMEDIA-2-before](https://user-images.githubusercontent.com/868127/114071142-b390e300-9898-11eb-836a-a182160d320e.png)


### After

![TMEDIA-2-after](https://user-images.githubusercontent.com/868127/114071164-ba1f5a80-9898-11eb-9df6-d0433168d3f1.png)


## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
